graphql-dgs-codegen-core/src/main/kotlin/com/netflix/graphql/dgs/codegen/generators/java/ClientApiGenerator.kt

```diff
@@ -173,13 +212,28 @@ class ClientApiGenerator(private val config: CodeGenConfig, private val document
             }
         }
 
+        if (it.inputValueDefinitions.isNotEmpty()) {
+            constructorBuilder.addCode(
+                """
+                |
+                |if(variableDefinitions != null) {
+                |   getVariableDefinitions().addAll(variableDefinitions);
+                |}
+                |
+                |if(variableReferences != null) {
+                |   getVariableReferences().putAll(variableReferences);
+                |}                      
+                """.trimMargin()
+            )
+        }
+
         val nameMethodBuilder = MethodSpec.methodBuilder("queryName")
             .addParameter(String::class.java, "queryName")
             .returns(ClassName.get("", "Builder"))
             .addModifiers(Modifier.PUBLIC)
             .addCode(
                 """
-                |this.queryName = queryName;
+                |this.queryName = queryName;                
```

Comment:
extra nit: revert extra spaces:
```suggestion
                |this.queryName = queryName;
```